src/jit/morph.cpp

```diff
@@ -19158,11 +19158,9 @@ Compiler::fgWalkResult Compiler::fgMarkAddrTakenLocalsPreCB(GenTree** pTree, fgW
                 // We may need to Quirk the storage size for this LCL_VAR
                 // some PInvoke signatures incorrectly specify a ByRef to an INT32
                 // when they actually write a SIZE_T or INT64
-                if (axc == AXC_Addr)
-                {
-                    comp->gtCheckQuirkAddrExposedLclVar(tree, fgWalkPre->parentStack);
-                }
+                comp->gtCheckQuirkAddrExposedLclVar(tree, fgWalkPre->parentStack);
```

Comment:
This is the main change that implements, Right?
"assume it escapes and that subsequent dereferences may be made at arbitrary offsets."

Reply:
Plus you added support to look into at Assignements and arguments to Calls properly.

Reply:
The main change is the one in the callback for `GT_CALL` and `GT_ASG`.

That change alters the address context that reaches here so I had to broaden this condition.